x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,529 @@
+certificate passwords that are stored in the
+<<secure-settings,secure settings>>, then you must complete a full cluster
+restart. {es} will not reload changes for passwords stored in the secure
+settings. We recommend <<use-rolling-restarts,rolling restarts>> when restarting your cluster.
+
+[[use-rolling-restarts]]
+.Rolling restarts are preferred
+****
+While it's possible to do an in-place update for security certificates, using
+a <<restart-cluster-rolling,rolling restart>> on your cluster is safer. An in-place update avoids some
+complications of a rolling restart, but incurs the following risks:
+
+* If you use PEM files, your certificate and key are in separate files. You
+must update both files _simultaneously_ or the node might experience a temporary
+period where it cannot establish new connections.
+* Updating the certificate and key does not automatically force existing
+connections to refresh. This means that even if you make a mistake, a node can
+seem like it's functioning but only because it still has existing connections.
+It's possible that a node will be unable to connect with other nodes, rendering
+it unable to recover from a network outage or node restart.
+****
+
+[[update-node-certs-same]]
+=== Update security certificates with the same CA
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+If your CA hasn't changed, you can just generate a new certificate, save it with
+the same name as your existing certificate, and replace the existing file.
+You'll need to generate new certificates and keys for both the transport layer
+and the HTTP layer.
+
+This procedure assumes that the you have access to the CA certificate and key
+that was originally generated (or otherwise held by your organization) and used
+to sign the certificate that is currently in use. It also assumes that the
+clients connecting to {es} on the HTTP layer are configured to trust the CA
+certificate.
+
+TIP: If your CA has changed, complete the steps in 
+<<update-node-certs-different,update security certificates with a different CA>>.
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
+Create a new PKCS#12 file that contains a certificate and key and replace
+the existing PKCS#12 file (such as `elastic-certificates.p12`) with the new
+certificate. 
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+security certificates that are currently in use. You'll use the same names
+for your new certificate.
++
+In this example, the keystore and truststore are pointing to different files.
+Your configuration might use the same file that contains the certificate and CA.
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.keystore.password: password
```

Comment:
xpack.security.transport.ssl.keystore.password is deprecated, do we want to point to it? Also, a few lines below we suggest users they add the password for the new keystore to the secure settings. The latter is best practice but it is maybe confusing to mention both in the same example?

Reply:
I didn't realize that this setting was deprecated, so thanks for the reminder! I'll remove it in favor of adding the password to the keystore in the secure settings.